### PR TITLE
[java-cdk] : Add UUID codecs to custom registry for MongoDB

### PR DESCRIPTION
**Pull request by akashkulk**
[java-cdk] : Add UUID codecs to custom registry while trying to parse mongodb events

We've chosen to select the STANDARD UUID codec provider here. We can only choose one and while this doesn't quite fix the root problem mentioned, it adds a reasonable default. 

The issue is that the UUID can be written via different clients - only running in the new packed mode will help with this. 

Part of https://github.com/airbytehq/oncall/issues/3346. Will publish mongo in a follow-up PR

### CODE REVIEW

**Comment by rodireich** [comment]
This looks good.
Is there no way to do cascading attempt, if not standard try legacy etc.
How will packed mode help? we still need to serialize a uuid out

**Review by rodireich — APPROVED** [review]

**Comment by rodireich** [comment]
Thinking about this issue:
* This issue is limited to incremental sync only if I understand correctly. How are other UUID subtypes getting handled during initial?
* Are there other types or subtypes we're unable to convert? either during initial or incremental

**Review by stephane-airbyte — CHANGES_REQUESTED** [review]
bump version to 0.4.2. Marius just pushed 0.4.1 to solve a merge conflict

**Comment by akashkulk** [comment]
> bump version to 0.4.2. Marius just pushed 0.4.1 to solve a merge conflict

Can you link me to the PR?

**Comment by stephane-airbyte** [comment]
> > bump version to 0.4.2. Marius just pushed 0.4.1 to solve a merge conflict
> 
> Can you link me to the PR?

https://github.com/airbytehq/airbyte/pull/32192

**Comment by stephane-airbyte** [comment]
@akashkulk I understand your request for a PR. Looks like 0.4.0's changelog should have been "fix compiler warnings", while 0.4.1s should have been "add 's3-destination' module to CDK"

File: airbyte-cdk/java/airbyte-cdk/README.md [hunk 1]
```diff
@@ -156,7 +156,9 @@ MavenLocal debugging steps:
...
 |:--------|:-----------|:-----------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| 0.4.0   | 2023-11-02 | [\#32050](https://github.com/airbytehq/airbyte/pull/32050) | Add 's3-destinations' CDK module.                                                                                                                              |
+| 0.4.2   | 2023-11-06 | [\#32119](https://github.com/airbytehq/airbyte/pull/32119) | Add STANDARD UUID code to MongoDB debezium handler.                                                                                                            |
+| 0.4.0   | 2023-11-02 | [\#32192](https://github.com/airbytehq/airbyte/pull/32192) | Add 's3-destinations' CDK module.                                                                                                                              |
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/README.md:160]: 0.4.1?

File: airbyte-cdk/java/airbyte-cdk/README.md [hunk 2]
```diff
@@ -156,7 +156,9 @@ MavenLocal debugging steps:
...
+| 0.4.0   | 2023-11-02 | [\#32050](https://github.com/airbytehq/airbyte/pull/32050) | Fix compiler warnings.                                                                                                                                         |
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/README.md:161]: thank you for adding this

**Review by stephane-airbyte — APPROVED** [review]

**Pull request merged**